### PR TITLE
NO-JIRA: bump openshift/api|client-go

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k 

/hold
the fixes in 2nd commit require figuring out how we want to handle them

### CODE REVIEW

File: pkg/manifest/manifest.go [hunk 1]
```diff
@@ -26,10 +26,14 @@ const (
...
-var knownFeatureSets = sets.String{}
+var knownFeatureSets = sets.Set[string]{}
 
 func init() {
-	for featureSet := range configv1.FeatureSets {
+	// map[FeatureSet]*FeatureGateEnabledDisabled{
+	// map[ClusterProfileName]map[FeatureSet]*FeatureGateEnabledDisabled
+	allFeatureSets := configv1.AllFeatureSets()
+	// TODO: should we read cluster profile?
+	for featureSet := range allFeatureSets[configv1.SelfManaged] {
```
Comment (deads2k) [inline pkg/manifest/manifest.go:36]: should iterate over both levels of the map and include all featureSets for this.

File: pkg/features/features.go [hunk 2]
```diff
@@ -88,13 +88,16 @@ func setFeatureGates(featureGatesMap map[string]bool, featureGates featuregate.M
 // InitializeFeatureGates should be called when your binary is starting with your featuregate instance and the list of
 // featuregates that your process will honor.
 func InitializeFeatureGates(featureGates featuregate.MutableFeatureGate, usedFeatures ...configv1.FeatureGateName) error {
```
Comment (deads2k) [inline pkg/features/features.go:90]: take clusterprofile as arg

Comment (soltysh) [reply pkg/features/features.go]: done

File: pkg/manifest/manifest.go (hunk 1 above)
Comment (soltysh) [reply pkg/manifest/manifest.go]: done

**Comment by deads2k** [comment]
/hold

the verify failure is real.

**Comment by soltysh** [comment]
/hold cancel
verify-deps - fixed

**Comment by p0lyn0mial** [comment]
/lgtm

the PR was tagged before but was failing on `verify-deps ` which has been fixed.

**Pull request merged**